### PR TITLE
OLS Workload

### PR DESCRIPTION
**Pull request by hayesgb**
This adds a stability test based on an OLS workflow that is reported to be stressful on the cluster.

### CODE REVIEW

File: tests/stability/test_ols.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import dask.array as da
+
+from ..utils_test import wait
+
+
+def test_ols(small_client, num_samples=int(1e8), num_coeffs=100, chunksize=int(1e6)):
```
Comment (jrbourbeau) [inline tests/stability/test_ols.py:6]: Nit: While defining `num_samples`, `num_coeffs`, and `chunksize` as keyword arguments seems to be working, this isn't typically done when writing `pytest`-style tests. Instead I'd recommend something like (note, I've not run linters on the code below):

```suggestion
def test_ols(small_client):
    num_samples = int(1e8)
    num_coeffs = 100
    chunksize = int(1e6)
```

which is more `pytest`-idiomatic. This should reduce the amount of thought future readers of this test will need.

**Review by jrbourbeau — COMMENTED** [review]
Thanks @hayesgb! 

Instead of creating a new `test_old.py` module for this test, let's move it to `tests/stability/test_array.py`, which seems more appropriate at the moment. If we end with a bunch of OLS-related tests, then we can pull things into a separate module. Since we're currently reusing clusters within modules, moving to `test_array.py` will also avoid the spinup time for an additional cluster. 

This test is taking ~14 minutes to run. Do we have a sense for what's problematic about this particular example? I wonder if we can reduce the runtime, while still preserving the usefulness as a stability test

**Comment by hayesgb** [comment]
cc: @gjoseph92

**Comment by hayesgb** [comment]
@jrbourbeau -- Thanks for the review.  I've tested this with `worker-saturation` on and the execution time drops to 10-11 minutes.  What are your thoughts on marking this as skipped until that's on be default?

**Comment by gjoseph92** [comment]
xref #252. Ideally I think we’d just add this and then see it start working
better when the default gets changed?

**Comment by hayesgb** [comment]
You're right @gjoseph92 .

After digging a little more, I've also learned this was run on a larger cluster, so I should be able to modify with `scaled_array_shape` to get the array / cluster size closer aligned to the original example.

**Comment by hayesgb** [comment]
This size array (relative to cluster size) is more closely represents the workload of interest.  At this scale, the. workload runs in about 100 seconds.  With `worker-saturation=1.0`, that drops to about 40.

**Comment by hayesgb** [comment]
Any additional concerns here @jrbourbeau ?

**Comment by ncclementi** [comment]
@hayesgb Would you mind reverting this and fixing all the CI commented text, and fixing the test suit to run?

**Pull request merged**